XenAdmin/Wizards/PatchingWizard/PatchingWizard.cs

```diff
@@ -281,6 +281,7 @@ private void CleanUploadedPatches(bool forceCleanSelectedPatch = false)
                 }
             }
 
+            list.RemoveAll(a => a == null);

```

Comment:
Instead of adding nulls and removing them afterwards, wouldn't it be better to check first and only add the action to the list if it is not null?

Reply:
Thanks @kc284, I've replace the commit.